### PR TITLE
Group By

### PR DESCRIPTION
**Pull request by ejoebstl**
Added
* group by support 
* aggregate_to_string aggregation function
* tests

### CODE REVIEW

File: src/ast/grouping.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+pub trait Groupable
+where
+    Self: Sized,
+{
+    /// Group by `self`
+    fn group(self) -> GroupByDefinition;
+}
+
+/// Convert the value into a group by definition.
+pub trait IntoGroupByDefinition {
```
Comment (pimeys) [inline src/ast/grouping.rs]: Why not using `From`?

```rust
impl<'a> From<&'a str> for GroupByDefinition {
    fn from(self) -> GroupByDefinition { ... }
}
```

File: src/ast/grouping.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+use crate::ast::{Column, DatabaseValue};
+
+pub type GroupByDefinition = (DatabaseValue);
```
Comment (pimeys) [inline src/ast/grouping.rs]: Why not just using `DatabaseValue` in `Grouping`? What does this type definition add to the code?

Comment (pimeys) [reply src/ast/grouping.rs]: Ah, now I get it. Limitation what can be used in the grouping!

File: src/ast/grouping.rs (hunk 1 above)
Comment (ejoebstl) [reply src/ast/grouping.rs]: I was just replicating the pattern used for `IntoOrderDefinition`. Should I change to `From`?

Comment (pimeys) [reply src/ast/grouping.rs]: No, leave it how it is. I remember now there was some limitations in place here why we needed our own trait.

**Pull request merged**